first step in exposing cli samples to solution gallery

Adam-it opened the pull request:
## 📜 first step in exposing cli samples to solution gallery

### 🎯 Aim

The aim of this PR is to modify one cli script sample in a way that will not break current gh pages and will also allow to expose script samples to solution gallery. 
Important step of this change is also to refactor the current folder structure that will allow as to add a dedicated assets folder for each script with their own sample.json and images

Adam-it commented:
@waldekmastykarz, @martinlingstuyl what do you think?
@martinlingstuyl could you guide me a bit how may I check if the current changes does not brake gh page for this sample 🤔🙏?

martinlingstuyl commented:
Hi @Adam-it, I think you should move the `sample.md` file to a directory with the name of the file, like you did. But then rename the actual file to `index.md`. Mkdocs will generate an `index.html` file for that, meaning that directory-urls will stay intact. 

https://www.mkdocs.org/user-guide/writing-your-docs/#index-pages

You should check if you need to change the `mkdocs.yml` > `nav` element, and the impact it has on other pages that link to your sample file. Normally, these are referenced using `path/to/sample-file.md` I think. This will no longer work, so it should be changed probably to `path/to/sample-file/index.md`. You could try without the `index.md`. 

The important thing is that Mkdocs should generate the links without `index.html` showing up anywhere. So that the page links are effectively exactly the same as they currently are in production.

Is that helpful?

@waldekmastykarz, any additions or changes?

Adam-it commented:
> Is that helpful?
> 

Yeeees 🤩. Your awesome 👍. thanks

Adam-it commented:
@martinlingstuyl so I modified the mkdocs.yml file and unfortunately as I tested it locally I had to add link directly to index.md file as when I just left the relative path it worked but the page was missing left side navigation 
![image](https://user-images.githubusercontent.com/58668583/174668976-5a91af65-c2c7-4984-bcde-ec69a956e0f5.png)

When I added link to index.md it works normally and we don't see the index.html file in the url 
![image](https://user-images.githubusercontent.com/58668583/174669053-68197cf2-f6ed-449a-8956-46d7349d5df6.png)

let me know if maybe I am not seeing something and we should reconsider this approach 🤔

martinlingstuyl commented:
No, this is fine, @Adam-it, so we need to update the `mkdocs.yml` file. That's OK as long as the urls are not referring to `index.html` when the application is serving/built.


How about links between pages? For example from the release notes page?

Adam-it commented:
> No, this is fine, @Adam-it, so we need to update the `mkdocs.yml` file. That's OK as long as the urls are not referring to `index.html` when the application is serving/built.
> 
> How about links between pages? For example from the release notes page?

I don't think any additional change will be needed as when comparing to the original site the link did not change at all and just typing it like this in browser works 
![image](https://user-images.githubusercontent.com/58668583/174670439-8fbe2f54-f992-4c92-aebe-16b38af1ef3a.png)

Adam-it commented:
@waldekmastykarz any additional comments here ? 
I am thinking to maybe have a try and merge this myself if that's fine with you😉

plamber commented:
Hi @Adam-it,
we can merge it together if you like.

Cheers

Adam-it commented:
> Hi @Adam-it,
> we can merge it together if you like. Just ping me
> 
> Cheers

That would be great. Will do.
Thanks a lot 👍

waldekmastykarz commented:
If the URLs and the current functionality is intact, it's good to go 🚀

Adam-it commented:
closing with manual merge
